spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindableProxyFactory.java

```diff
@@ -102,130 +72,12 @@ public void setBeanFactory(BeanFactory beanFactory) throws BeansException {
-						BindableProxyFactory.this.inputs.put(name, new ChannelHolder(inputChannel, true));
-					}
-					else {
-						if (inputChannelType.isAssignableFrom(sharedChannel.getClass())) {
-							BindableProxyFactory.this.inputs.put(name, new ChannelHolder(sharedChannel, false));
-						}
-						else {
-							// handle the special case where the shared channel is of a different nature
-							// (i.e. pollable vs subscribable) than the target channel
-							final MessageChannel inputChannel = createMessageChannel(inputChannelType);
-							if (isPollable(sharedChannel.getClass())) {
-								bridgePollableToSubscribableChannel(sharedChannel,
-										inputChannel);
-							}
-							else {
-								bridgeSubscribableToPollableChannel(
-										(SubscribableChannel) sharedChannel, inputChannel);
-							}
-							BindableProxyFactory.this.inputs.put(name, new ChannelHolder(inputChannel, false));
-						}
-					}
-				}
-
-				Output output = AnnotationUtils.findAnnotation(method, Output.class);
-				if (output != null) {
-					String name = BindingBeanDefinitionRegistryUtils.getChannelName(
-							output, method);
-					Class<?> messageChannelType = method.getReturnType();
-					MessageChannel sharedChannel = locateSharedChannel(name);
-					if (sharedChannel == null) {
-						MessageChannel outputChannel = createMessageChannel(messageChannelType);
-						BindableProxyFactory.this.outputs.put(name, new ChannelHolder(outputChannel, true));
-					}
-					else {
-						if (messageChannelType.isAssignableFrom(sharedChannel.getClass())) {
-							BindableProxyFactory.this.outputs.put(name, new ChannelHolder(sharedChannel, false));
-						}
-						else {
-							// handle the special case where the shared channel is of a different nature
-							// (i.e. pollable vs subscribable) than the target channel
-							final MessageChannel outputChannel = createMessageChannel(messageChannelType);
-							if (isPollable(messageChannelType)) {
-								bridgePollableToSubscribableChannel(outputChannel,
-										sharedChannel);
-							}
-							else {
-								bridgeSubscribableToPollableChannel(
-										(SubscribableChannel) outputChannel,
-										sharedChannel);
-							}
-							BindableProxyFactory.this.outputs.put(name, new ChannelHolder(outputChannel, false));
-						}
-					}
-				}
-			}
-		});
-	}
-
-	private MessageChannel locateSharedChannel(String name) {
-		return this.sharedChannelRegistry != null ? this.sharedChannelRegistry.get(getNamespacePrefixedChannelName(name)) : null;
-	}
-
-	private String getNamespacePrefixedChannelName(String name) {
-		return this.channelNamespace + "." + name;
-	}
-
-	private void bridgeSubscribableToPollableChannel(SubscribableChannel sharedChannel,
-			MessageChannel inputChannel) {
-		sharedChannel.subscribe(new MessageChannelBinderSupport.DirectHandler(
-				inputChannel));
+		this.channelFactory = beanFactory.getBean(ChannelFactory.class);
```

Comment:
No, this isn't needed. I was trying something else and forgot to remove this and use `Autowired`. will fix it.